src/generate_form.jsx

```diff
@@ -0,0 +1,285 @@
+const config = {
+	title: "HCC Calculator",
+	info: "A post-transplant hepatocellular carcinoma (HCC) recurrence calculator using a machine-learning algorithm mapped on pre-operative patient and tumor characteristics",
+	roundoff: 2
+}
+
+const variables = [{
+	label: "Pre-transplant neutrophil count",
+	name: "nCnt",
+	type: "Float",
+	display: "Input",
+	weight: 0.1
+},{
+	label: "Number of bridging treatments",
+	name: "nBT",
+	type: "Integer",
+	display: "Dropdown",
+	upper: 8,
+	weight: 0.2
+},{
+	label: "Number of PEI treatments",
+	name: "nPEI",
+	type: "Integer",
+	display: "Dropdown",
+	upper: 12,
+	weight: 0.3
+},{
+	label: "Size of largest lesion at time of listing",
+	name: "lSize",
+	type: "Float",
+	display: "Input",
+	weight: 0.4
+},{
+	label: "Total tumor diameter at the time of listing",
+	name: "tDiameter",
+	type: "Float",
+	display: "Input",
+	weight: 0.5
+}];
+
+import { Typography, Container, Grid, DialogContentText } from '@material-ui/core';
+import React from "react";
+import ReactDOM from "react-dom";
+import { makeStyles, withStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+
+import Button from '@material-ui/core/Button';
+
+import InfoIcon from '@material-ui/icons/Info';
+import Tooltip from '@material-ui/core/Tooltip';
+
+import Dialog from '@material-ui/core/Dialog';
+import DialogContent from '@material-ui/core/DialogContent';
+
+const useSelectStyles = makeStyles(theme => ({
+  root: {
+    display: 'flex',
+    flexWrap: 'wrap',
+  },
+  select: {
+    minWidth: 320
+  },
+}));
+
+const useTextFieldStyles = makeStyles(theme => ({
+	textfield: {
+		minWidth: 320
+	}
+}));
+
+const useButtonStyles = makeStyles(theme => ({
+  button: {
+    margin: theme.spacing(1),
+  },
+  input: {
+    display: 'none'
+  },
+}));
+
+const useBlackTooltipStyles = makeStyles(theme => ({
+  tooltip: {
+    color: "white",
+    backgroundColor: "black"
+  }
+}));
+
+function Field(props) {
+		const select_classes = useSelectStyles();
+		const textfield_classes = useTextFieldStyles();
+
+		const [val, setVal] = React.useState('');
+
+		function handleChange(event) {
+			setVal(event.target.value);
+		}
+		return (
+			<Grid item>
+			<Grid container direction="column" spacing={0}>
+				<Grid item>
+					<Typography variant="subtitle1" gutterBottom>
+						{props.metadata.label}
+					</Typography>
+				</Grid>
+				<Grid item>
+				{
+					props.metadata.display == "Input" ? 
+					(<TextField
+						className={textfield_classes.textfield}
+        				id={props.metadata.name}
+						type="number"
+						onChange={handleChange}
+						error={props.err}
+        				margin="dense"
+        				width="auto"/>)
+					:
```

Comment:
I initially had an if statement for that reason, but the compiler would complain about it. Since then I've changed the transpiler, so it may work now.